### PR TITLE
Add a section on Chef server credentials

### PR DESCRIPTION
**Pull request by stevendanna**
This provides users with a single place to read about the current
state of credentials management in Chef server and the related
add-ons.

Signed-off-by: Steven Danna <steve@chef.io>

### CODE REVIEW

File: chef_master/source/server_security.rst [hunk 1]
```diff
@@ -268,6 +268,55 @@ To regenerate SSL certificates:
...
+     - *none*
+
+These newer add-ons will also write all of their secrets to ``/etc/opscode/private-chef-secrets.json``. Older versions of the add-ons will still write their configuration to locations in ``/etc`` and ``/var/opt``.
+
+/etc/opscode/private-chef-secrets.json
++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+``/etc/opscode/private-chef-secrets.json``'s default permissions only allow the root user to read or write the file. This file contains all of the secrets for access to the Chef server's underlying data stores, and thus access to it should be restricted to trusted users.
+
+While the file does not contain passwords in plaintext, it is not safe to share with untrusted users. The format of the secrets file, allows deployments of Chef server to conform to regulations requiring that no sensitive data should appear in plain text in configuration files; however, it does not make the file meaningfully more secure.
```
Comment (blakestier) [inline chef_master/source/server_security.rst]: no comma here. `secrets file, allows` -> `secrets file allows`

Comment (blakestier) [reply chef_master/source/server_security.rst]: comma after `regulations` (`regulations, requiring`)

File: chef_master/source/server_security.rst [hunk 2]
```diff
@@ -268,6 +268,55 @@ To regenerate SSL certificates:
...
+
+DRBD and Keepalived
++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+In the DRBD-based HA configuration, Chef server will render passwords for keepalived and DRBD to configuration files in ``/var/opt/opscode``.
```
Comment (blakestier) [inline chef_master/source/server_security.rst:319]: `keepalive`? (vs `keepalived`)?

File: chef_master/source/server_security.rst [hunk 3]
```diff
@@ -268,6 +268,55 @@ To regenerate SSL certificates:
...
+   * - Chef Manage
+     - 2.5.0
+   * - Push Jobs Server
+     - 2.2.0
+   * - Reporting
+     - 1.7.0
+   * - Analytics
+     - *none*
+
+These newer add-ons will also write all of their secrets to ``/etc/opscode/private-chef-secrets.json``. Older versions of the add-ons will still write their configuration to locations in ``/etc`` and ``/var/opt``.
```
Comment (blakestier) [inline chef_master/source/server_security.rst:307]: Will older versions also write to `private-chef-secrets`? Maybe "will still only write"

**Comment by blakestier** [comment]
Couple grammar nitpicks and clarifying questions. Otherwise, looks great!

File: chef_master/source/server_security.rst (hunk 2 above)
Comment (juliandunn) [reply chef_master/source/server_security.rst:319]: `keepalived` is a program (the "keep alive daemon")

File: chef_master/source/server_security.rst (hunk 3 above)
Comment (stevendanna) [reply chef_master/source/server_security.rst:307]: Yes, older versions write to private-chef-secrets.json and many other locations in /etc :D

File: chef_master/source/server_security.rst (hunk 1 above)
Comment (stevendanna) [reply chef_master/source/server_security.rst]: I'v fixed the first comma, but I feel like the comma after regulations would change the meaning here.

**Comment by stevendanna** [comment]
@blakestier Thanks for the review, pushed some changes and responded inline to the q's.

File: chef_master/source/server_security.rst (hunk 2 above)
Comment (blakestier) [reply chef_master/source/server_security.rst:319]: I thought that might be the case! Thanks :)

File: chef_master/source/server_security.rst (hunk 1 above)
Comment (blakestier) [reply chef_master/source/server_security.rst]: Ah, I see your intention, here. Totally right in this context 👍

File: chef_master/source/server_security.rst (hunk 3 above)
Comment (blakestier) [reply chef_master/source/server_security.rst:307]: Okay-- that wasn't obvious to me but it's totally non-blocking.

**Comment by blakestier** [comment]
Looks good to me. 🚢

**Comment by kagarmoe** [comment]
Thanks for all your hard work!  Great job with the version annotations!

**Comment by kagarmoe** [comment]
gif-keyboard-17080088247670038979

**Pull request merged**